src/components/range-inputs.tsx

```diff
@@ -0,0 +1,303 @@
+import '@isrd-isi-edu/chaise/src/assets/scss/_range-input.scss';
+import { useState, useRef } from 'react';
+import { windowRef } from '@isrd-isi-edu/chaise/src/utils/window-ref';
+import { ClearInputBtn } from '@isrd-isi-edu/chaise/src/components/clear-input-btn';
+
+
+const INTEGER_REGEXP = /^\-?\d+$/;
+
+const FLOAT_REGEXP = /^\-?(\d+)?((\.)?\d+)?$/;
+
+const TIME_FORMAT = 'YYYY-MM-DDTHH:mm';
+
+const DATE_FORMAT = 'YYYY-MM-DD';
+
+/**
+ * an object for mapping the error codes to the error messages
+ */
+const errorMsgMap: {
+    [key: string]: string;
+} = {
+    'empty': 'Please enter either From or To value',
+    'range': 'From value cannot be greater than the To value',
+    'int': 'Please enter an integer value',
+    'float': 'Please enter a decimal value',
+    'date': 'Please enter a date value in YYYY-MM-DD format',
+    'timestamp': 'Please enter a time value in YYYY-MM-DD HH:MM format'
+};
+
+type DateTimePickerProps = {
```

Comment:
Can you add jsdoc comments for each prop being added? Explain what the property is for and any restrictions it might have.

similar to how Ravish had done it here:
https://github.com/informatics-isi-edu/chaise/blob/react/src/components/facet-header.tsx#L12-L37